File: Data Structures & Algorithms/two-integer-sum/submission-5.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        index_map = {}
        for i, n in enumerate(nums):
            comp = target - n
            if comp in index_map:
                return [index_map[comp], i]
            else:
                index_map[n] = i
                
        

File: Data Structures & Algorithms/two-integer-sum-ii/submission-5.js

class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(numbers, target) {
        let l = 0,
            r = numbers.length-1;

        while(l<r){
            const temp = numbers[l]+numbers[r];
            if(temp > target) r-=1;
            else if (temp === target) return [l+1, r+1];
            else l+=1;
        }
    }
}
